src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp

```diff
@@ -1142,95 +789,62 @@ winrt::hstring CascadiaSettings::ApplicationDisplayName()
-        UINT versionLen{ 0 };
-        const auto localizedVersionName{ wil::str_printf<std::wstring>(L"\\StringFileInfo\\%04x%04x\\ProductVersion",
-                                                                       pVarLocalization->language ? pVarLocalization->language : 0x0409, // well-known en-US LCID
-                                                                       pVarLocalization->codepage) };
-        THROW_IF_WIN32_BOOL_FALSE(VerQueryValueW(versionBuffer.get(), localizedVersionName.c_str(), reinterpret_cast<void**>(&pProductVersion), &versionLen));
-        return { pProductVersion };
+
+        const auto versionBuffer = std::make_unique<std::byte[]>(size);
+        THROW_IF_WIN32_BOOL_FALSE(GetFileVersionInfoExW(0, filename.get(), 0, size, versionBuffer.get()));
+
+        VS_FIXEDFILEINFO* info = nullptr;
+        UINT infoSize = 0;
+        THROW_IF_WIN32_BOOL_FALSE(VerQueryValueW(versionBuffer.get(), L"\\", reinterpret_cast<void**>(&info), &infoSize));
+        THROW_HR_IF(E_UNEXPECTED, infoSize < sizeof(*info));
+
+        return formatVersion(
+            HIWORD(info->dwProductVersionMS),
+            LOWORD(info->dwProductVersionMS),
+            HIWORD(info->dwProductVersionLS),
+            LOWORD(info->dwProductVersionLS));
     }
     CATCH_LOG();
 
     return RS_(L"ApplicationVersionUnknown");
 }
 
 // Method Description:
-// - Forces a refresh of all default terminal state. This hits the registry to
-//   read off the disk, so best to not do it on the UI thread.
+// - Determines if we're on an OS platform that supports
+//   the default terminal handoff functionality.
 // Arguments:
 // - <none>
 // Return Value:
-// - <none> - Updates internal state
-void CascadiaSettings::RefreshDefaultTerminals()
-{
-    _defaultTerminals.Clear();
-
-    for (const auto& term : Model::DefaultTerminal::Available())
-    {
-        _defaultTerminals.Append(term);
-    }
-
-    _currentDefaultTerminal = Model::DefaultTerminal::Current();
-}
-
-// Helper to do the version check
-static bool _isOnBuildWithDefTerm() noexcept
+// - True if OS supports default terminal. False otherwise.
+bool CascadiaSettings::IsDefaultTerminalAvailable() noexcept
 {
-    OSVERSIONINFOEXW osver{ 0 };
+    OSVERSIONINFOEXW osver{};
     osver.dwOSVersionInfoSize = sizeof(osver);
     osver.dwBuildNumber = 21359;
 
     DWORDLONG dwlConditionMask = 0;
     VER_SET_CONDITION(dwlConditionMask, VER_BUILDNUMBER, VER_GREATER_EQUAL);
 
-    return VerifyVersionInfoW(&osver, VER_BUILDNUMBER, dwlConditionMask);
-}
-
-// Method Description:
-// - Determines if we're on an OS platform that supports
-//   the default terminal handoff functionality.
-// Arguments:
-// - <none>
-// Return Value:
-// - True if OS supports default terminal. False otherwise.
-bool CascadiaSettings::IsDefaultTerminalAvailable() noexcept
-{
-    // Cached on first use since the OS version shouldn't change while we're running.
-    static bool isAvailable = _isOnBuildWithDefTerm();

```

Comment:
This function isn't called particularly often. (Once a day maybe? 😄)
And the test is just a single call to a Win32 API so it should be fine. So I removed the caching because it consumes valuable static space in the executable.